concent_api/core/templates/admin/payment_report.html

```diff
@@ -0,0 +1,155 @@
+{% extends "admin/base_site.html" %}
+
+{% load static %}
+{% load admin_tags %}
+{% load admin_urls %}
+
+{% block title %}Payment report | {{ site_title|default:_('Django site admin') }}{% endblock %}
+
+{% block content %}
+<style>
+table, th, td {
+  border: 1px solid black;
+  border-collapse: collapse;
+}
+th, td {
+  padding: 5px;
+  text-align: left;
+}
+br {
+    padding: 5px;
+}
+</style>
+
+    {% for pair_report in payment_report.pair_reports %}
+    <table style="width:100%">
+    <caption>Pair report {{ forloop.counter }}</caption>
+        <tr>
+            <th>Provider:</th>
+            <td colspan="2">
+                {% for address in  pair_report.provider_ethereum_addresses %}
+                    <a href="https://{{ eth_network }}etherscan.io/address/{{ address }}">{{ address }}</a>,
+                {% endfor %}
+                public key: {{ pair_report.provider.public_key }}
+            </td>
+        </tr>
+        <tr>
+            <th>Requestor:</th>
+            <td colspan="2">
+                {% for address in  pair_report.requestor_ethereum_addresses %}
+                    <a href="https://{{ eth_network }}etherscan.io/address/{{ address }}">{{ address }}</a>,
+                {% endfor %}
+                public key: {{ pair_report.requestor.public_key }}
+            </td>
+        </tr>
+
+        {#  Pending Responses table  #}
+        <tr>
+            <th rowspan="3">PendingResponses</th>
+            <th>SubtaskResultsSettled</th>
+            <td>
+            {% for subtask_results_settled in pair_report.pending_responses.subtask_results_settled %}
+                <p><a href="{% url 'admin:core_pendingresponse_change' subtask_results_settled.id %}">
```

Comment:
You should create partial template for this and include it three times.

Reply:
I'm not sure how that suppose to look. Could You give me an example?

Reply:
https://docs.djangoproject.com/en/2.1/ref/templates/builtins/#include

Reply:
@Jakub89 You simply extract repeating part to separate html file, and use `include` tag to attach it.

Reply:
Done